### PR TITLE
Initial OpenCost security policy

### LINKED ISSUE #1845: OpenCost needs a public security policy
A SECURITY.md documenting the security policy, supported releases, how to report potential issues, and a disclosure policy are standard for CNCF projects.

https://clomonitor.io/docs/topics/checks/#security-policy

### PR DESCRIPTION
**Pull request by mattray**
This PR is meant to provide a starting point for the OpenCost security policy which documents supported releases, how to report potential issues, and a disclosure policy as standard for CNCF projects.

Before merging, "Private vulnerability reporting" will need to be turned on for the GitHub repository. Enabling GitHub code scanning will be added later.

Once merged, this closes https://github.com/opencost/opencost/issues/1845 and PRs will be opened for the [opencost-helm-chart](https://github.com/opencost/opencost-helm-chart/) and [opencost-website](https://github.com/opencost/opencost-website/) to ensure they have similar policies.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+For example, if `v1.102.0` is the most recent stable version, we will address security updates for `v1.101.0` and later. Once `v1.103.0` is released, we will no longer provide updates for `v1.101.x` releases.
+
+## Reporting a Vulnerability
+
+The OpenCost project has enabled [Private vulnerability reporting](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability) for our repositories which allows for direct reporting of issues to administrators and maintainers in a secure fashion. Please include a thorough description of the issue, the steps you took to create the issue, affected versions, and, if known, mitigations for the issue. The team will help diagnose the severity of the issue and determine how to address the issue. Issues deemed to be non-critical will be filed as GitHub issues. Critical issues will receive immediate attention and be fixed as quickly as possible.
+
+### Kubecost Bug Bounty
+
+Kubecost offers a Bug Bounty program that pays $250 USD for unique accepted security bug reports submitted to vulnerability-report@kubecost.com.
```
Comment (AjayTripathy) [inline SECURITY.md]: I would clarify that these are not for publicly available CVEs

**Review by AjayTripathy — APPROVED** [review]
One comment but then feel free to merge.

**Comment by mattray** [comment]
@AjayTripathy updated. If that's good, I'll make a similar PR to update Kubecost's bug bounty too: https://docs.kubecost.com/troubleshooting/bug-bounty-program

**Comment by AjayTripathy** [comment]
Sgtm

**Pull request merged**